pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/datasource/DataSource.java

```diff
@@ -76,6 +76,12 @@
   @Nullable
   TextIndexReader getFSTIndex();
 
+  /**
+   * Returns the Native FST index for the column if exists, or {@code null} if not.
+   */
+  @Nullable
+  TextIndexReader getNativeFSTIndex();
```

Comment:
Since native FST and lucene FST are both FST, just different implementation, and I don't think they should co-exist, let's reuse the same `getFSTIndex()` for both of them. We shouldn't need to change any code on the query execution side. The loader should be able to tell which version of FST exists and load it as the FST index.

Reply:
The problem is during creation -- there is no way to specify at a per field level as to which FST type needs to be used. Hence the need for the extra index